include/range/v3/detail/config.hpp

```diff
@@ -291,6 +291,42 @@
 #define RANGES_GCC_BROKEN_CUSTPOINT
 #endif
 
+#define RANGES_CXX_INLINE_VARIABLES_11 0
+#define RANGES_CXX_INLINE_VARIABLES_14 0
+#define RANGES_CXX_INLINE_VARIABLES_17 201606
+#ifndef RANGES_CXX_INLINE_VARIABLES
+
+#ifdef __cpp_inline_variables // TODO: fix this if SD-6 picks another name
+#define RANGES_CXX_INLINE_VARIABLES __cpp_inline_variables
+#elif defined(__clang__) && \
+    (__clang__major__ > 3 || __clang_major__ == 3 && __clang_minor__ == 9) && \
+    __cplusplus > 201402L
+// TODO: remove once clang defines __cpp_inline_variables (or equivalent)
+#define RANGES_CXX_INLINE_VARIABLES RANGES_CXX_INLINE_VARIABLES_17
+#else
+#define RANGES_CXX_INLINE_VARIABLES RANGES_CXX_FEATURE(INLINE_VARIABLES)
+#endif  // __cpp_inline_variables
+
+#endif  // RANGES_CXX_INLINE_VARIABLES
+
+#if RANGES_CXX_INLINE_VARIABLES < RANGES_CXX_INLINE_VARIABLES_17
+#define RANGES_INLINE_VARIABLE_(type, name, inline_)      \
+    inline_ namespace                                     \
+    {                                                     \
+        constexpr auto& name = static_const<type>::value; \
+    }                                                     
+
+#else  // RANGES_CXX_INLINE_VARIABLES >= RANGES_CXX_INLINE_VARIABLES_17
+#define RANGES_INLINE_VARIABLE_(type, name, inline_) \
+    namespace function_objects                       \
```

Comment:
My vague recollection is that some supported versions of GCC only worked *with* `inline`, and some supported versions only worked *without* `inline`, so that we could not simply tack `inline` onto all of the unnamed namespaces. All supported variations have Travis coverage, so feel free to make the change and see if it makes it through Travis.